blocks/gnav/gnav.js

```diff
@@ -1,287 +1,414 @@
+  decorateMainNav = () => {
+    const mainLinks = this.body.querySelectorAll('h2 > a');
+    if (mainLinks.length > 0) {
+      return this.buildMainNav(mainLinks);
+    }
+    return null;
   }
-}
 
-function getRelativeURL(href) {
-  const url = new URL(href, window.location);
-  if (url.hostname.includes('blog.adobe.com')
-    || url.hostname.includes('.page')
-    || url.hostname.includes('.live')
-    || url.hostname.includes('localhost')) {
-    return (url.pathname);
+  buildMainNav = (navLinks) => {
+    const mainNav = createEl({ tag: 'div', className: 'gnav-mainnav' });
+    navLinks.forEach((navLink, idx) => {
+      const navItem = createEl({ tag: 'div', className: 'gnav-navitem' });
+
+      const menu = navLink.closest('div');
+      menu.querySelector('h2').remove();
+      navItem.appendChild(navLink);
+
+      if (menu.childElementCount > 0) {
+        const id = `navmenu-${idx}`;
+        menu.id = id;
+        navItem.classList.add('has-Menu');
+        navLink.setAttribute('role', 'button');
+        navLink.setAttribute('aria-expanded', false);
+        navLink.setAttribute('aria-controls', id);
+
+        const decoratedMenu = this.decorateMenu(navItem, navLink, menu);
+        navItem.appendChild(decoratedMenu);
+      }
+      mainNav.appendChild(navItem);
+    });
+    return mainNav;
   }
-  return (href);
-}
 
-function isSelected(navItem) {
-  if (navItem.submenu) {
-    const matches = navItem.submenu.filter((e) => {
-      const navpath = new URL(e.href, window.location).pathname;
-      return (navpath === window.location.pathname);
+  decorateMenu = (navItem, navLink, menu) => {
+    menu.className = 'gnav-navitem-menu';
+    const childCount = menu.childElementCount;
+    if (childCount === 1) {
+      menu.classList.add('small-Variant');
+    } else if (childCount === 2) {
+      menu.classList.add('medium-Variant');
+    } else if (childCount >= 3) {
+      menu.classList.add('large-Variant');
+    }
+    navLink.addEventListener('click', (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      this.toggleMenu(navItem);
     });
-    if (matches.length) return (true);
+    return menu;
   }
-  if (navItem.href) {
-    const navpath = new URL(navItem.href, window.location).pathname;
-    return (navpath === window.location.pathname);
+
+  decorateSearch = () => {
+    const searchBlock = this.body.querySelector('.search');
+    if (searchBlock) {
+      const label = searchBlock.querySelector('p').textContent;
+      const advancedLink = searchBlock.querySelector('a');
+      const searchEl = createEl({ tag: 'div', className: 'gnav-search' });
+      const searchBar = this.decorateSearchBar(label, advancedLink);
+      const searchButton = createEl({
+        tag: 'button', className: 'gnav-search-button', html: SEARCH_ICON, attributes: { 'aria-label': label },
+      });
+
+      searchButton.addEventListener('click', async () => {
+        if (!this.onSearchInput) {
```

Comment:
this seems to be used in two places, maybe it's a good idea to have it as a separate method
https://github.com/adobe/business-website/pull/104/files#diff-9046cde4989d868db3e34286dbdf9a23a0966027cbc3cd7c10cb5925b3780bb5R83

Reply:
Fixed.